diagrams/media.create.sequenceDiagram.plantuml

```diff
@@ -0,0 +1,19 @@
+@startuml create media
+hide footbox
+
+
+participant frontend
+participant posts
+participant mediaService
+participant uploadService
+
+
+frontend -> mediaService: Binary file + id
+mediaService -> mediaService: store in disk (buffer)
+mediaService -> frontend: ack
```

Comment:
Shouldn't we send the `ack` always first since we are working `asynchronously` and <strong> <u> after that </u> </strong>  we store it in file system as per current task?

Reply:
what if, for any reason, the file wasn't stored successfully in disk. We should respond with ack or (received) after we ensure that we have the file in our system

Reply:
I agree with you, but this opens up a question for me.
For the same reason, what if when a user creates a new post, why do we respond with `ack` i.e. working async and at the same time we're not sure if the file will get created or not?

Reply:
I suggest leaving this whenever we have a meet to discuss it.